daisi/docs/minhton-ns3/logging.md

```diff
@@ -1,3 +1,74 @@
+
+## Views
+
+Since the tables partially contain values that require further context, we also provide some database views for easier understanding of the logged information. The views are named after their originating tables:
+
+- **viewMinhtonNodeState**
+- **viewMinhtonTraffic**
+- **viewRoutingInfo**
+
+## Further Information
+
+### MinhtonTraffic
+
+Mode means either that the message got sent or received.
+
+| Mode | Interpretation |
+|------|----------------|
+| 0    | Receiving      |
+| 1    | Sending        |
+
+Every other field represents information about the content of the message. Sender and target must always be set. Depending on the message type the other fields might also be set. In most cases only two additional fields for nodes are necessary. Other information like a vector of nodes will be written into the Content field as a string.
+
+### SearchTest
+
+| Search Exact Status | Value | Interpretation                  |
+|---------------------|-------|---------------------------------|
+| Start               | 0     | Sender starts sending           |
+| Hop                 | 1     | Intermediate step               |
+| Success             | 2     | Target reached                  |
+| Failure             | 3     | Target not found / non-existent |
+
+When executing a search test, search exact messages will be sent from certain nodes to certain other nodes. The search exact messages contain empty messages as a query without functionality.
+
+The sender is the node who initializes the search exact messages and wants to send the message to the target / destination. The hop nodes are nodes, who are not the sender and not the target, but intermediate steps along the way towards the destination.
+
+The status of an entry can be either Start, Hop, Success or Failure. A message is only successfully passed to the target if at exactly one entry with the Success status from sender to target exists. The message passing has definitely failed if there is an entry with the Failure status. Everything else should not happen and would be undefined behavior. Entries with the Start status should not be count as messages passing through the network, because it shows the initial processing and forwarding of the message still within the sender node.
+
+When a Failure status occurs this can either mean that the target node does not exist (which might happen), or that there is an error in the algorithm and the forwarding does not work correctly.
+
+### RoutingInfo
+
+To evaluate the routing information logged in the RoutingInfo table, is is necessary to calculate the latest valid entries at a given point in time, because the neighbor relationships are constantly changing. For example while just building a network of 7 nodes with fanout 2, the adjacents of root change multiple times. First 0:0 has the left adjacent 1:0, then later 2:1. Here the left adjacent neighbor relationship will just be overwritten with the new adjacent left.
```

Comment:
```suggestion
To evaluate the routing information logged in the RoutingInfo table, it is necessary to calculate the latest valid entries at a given point in time, because the neighbor relationships are constantly changing. For example while just building a network of 7 nodes with fanout 2, the adjacents of root change multiple times. First 0:0 has the left adjacent 1:0, then later 2:1. Here the left adjacent neighbor relationship will just be overwritten with the new adjacent left.
```